api/v1alpha1/common_types.go

```diff
@@ -437,6 +444,39 @@ func InitializeHorizontalPodAutoscalerSpec(spec *HorizontalPodAutoscalerSpec, de
 	return spec
 }
 
+type DeploymentStrategySpec struct {
+	// Type of deployment. Can be "Recreate" or "RollingUpdate". Default is RollingUpdate.
+	// +optional
+	Type appsv1.DeploymentStrategyType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type,casttype=DeploymentStrategyType"`
```

Comment:
this needs to be a pointer, per API conventions

Reply:
and why the protobuff tags?

Reply:
> and why the protobuf tags?

Fixed in https://github.com/3scale-ops/saas-operator/pull/257/commits/0d042ac5d57ea3991584684484945a5eaa999696

Reply:
> this needs to be a pointer, per API conventions

It doesn't use a pointer to match https://pkg.go.dev/k8s.io/api/apps/v1#DeploymentStrategy